From: Thomas Renninger <trenn@suse.de>
Subject: Allocate memory GFP_ATOMIC in irq_disbled (suspend) case
Patch-mainline: 
References: 153062

This does not solve the problem fully, therefore added with
trenn tagged for now.

Signed-off-by: Thomas Renninger <trenn@suse.de>

Index: linux-2.6.15/drivers/acpi/osl.c
===================================================================
--- linux-2.6.15.orig/drivers/acpi/osl.c
+++ linux-2.6.15/drivers/acpi/osl.c
@@ -1252,7 +1252,7 @@ acpi_status acpi_os_release_object(acpi_
 
 void *acpi_os_acquire_object(acpi_cache_t * cache)
 {
-	void *object = kmem_cache_alloc(cache, GFP_KERNEL);
+	void *object = kmem_cache_alloc(cache, GFP_ATOMIC);
 	WARN_ON(!object);
 	return object;
 }
